docs/standard/assembly/reference-assemblies.md

```diff
@@ -0,0 +1,74 @@
+
+You can generate reference assemblies:
+
+- In an MSBuild project, by using the [`ProduceReferenceAssembly` project property](/visualstudio/msbuild/common-msbuild-project-properties).
+- When compiling program from command line, by specifiying `-refonly` ([in C#](../../csharp/language-reference/compiler-options/refonly-compiler-option.md), [in Visual Basic](../../visual-basic/reference/command-line-compiler/refonly-compiler-option.md) ) or `-refout` ([in C#](../../csharp/language-reference/compiler-options/refout-compiler-option.md), [in Visual Basic](../../visual-basic/reference/command-line-compiler/refout-compiler-option.md)) compiler options
+- When using the Roslyn API, by setting <xref:Microsoft.CodeAnalysis.Emit.EmitOptions.EmitMetadataOnly?displayProperty=nameWithType> to `true` and <xref:Microsoft.CodeAnalysis.Emit.EmitOptions.IncludePrivateMembers?displayProperty=nameWithType> to `false` in an object passed to the <xref:Microsoft.CodeAnalysis.Compilation.Emit%2A?displayProperty=nameWithType> method.
+
+If you want to distribute reference assemblies with NuGet packages, you must include them in the *ref\\* subdirectory under the package directory instead of in the *lib\\* subdirectory used for implementation assemblies.
+
+## Reference assemblies structure
+
+Reference assemblies are an expansion of the related concept, *metadata-only assemblies*. Metadata-only assemblies have their method bodies replaced with a single `throw null` body, but include all members except anonymous types. The reason for using `throw null` bodies (as opposed to no bodies) is so that PEVerify can run and pass (thus validating the completeness of the metadata).
+
+Reference assemblies further remove metadata (private members) from metadata-only assemblies:
+
+- A reference assembly only has references for what it needs in the API surface. The real assembly may have additional references related to specific implementations. For instance, the reference assembly for `class C { private void M() { dynamic d = 1; ... } }` does not reference any types required for `dynamic`.
+- Private function-members (methods, properties, and events) are removed in cases where their removal doesn't observably impact compilation. If there are no [InternalsVisibleTo](xref:System.Runtime.CompilerServices.InternalsVisibleToAttribute) attributes, internal function members are also removed.
+
+The metadata in reference assemblies continues to retain information about the following:
+
+- All types, including private and nested types.
+- All attributes, even internal ones.
+- All virtual methods.
+- Explicit interface implementations. 
+- Explicitly implemented properties and events, because their accessors are virtual.
+- All fields of structures. 
```

Comment:
See above; we don't preserve private struct fields. Rather, we emit dummy fields matching their characteristic.

Reply:
See this discussion: https://github.com/dotnet/docs/pull/14393#discussion_r331725324 What i stated there is correct for reference assemblies generated by Roslyn compilers. I've noted in the end of the section that .NET Core reference assemblies are special case, but i don't think we should document all details like dummy struct fields. This might be implementation detail because this wasn't previously documented anywhere. Also the rules are pretty complicated: from my observations, it inserts dummy object field if the struct has at least one reference type field, or dummy int32 field if at least one value type field. But this is probably not of big interest for most readers; what only matters is that compiler won't treat structs incorrectly. But if these really needs to be included, let me know.